R/zchunk_L2322.Fert.R

```diff
@@ -76,272 +76,456 @@ module_energy_L2322.Fert_DISABLED <- function(command, ...) {
+      rename(stub.technology = technology) ->
+      L2322.StubTech_Fert
+
+    # L2322.GlobalTechShrwt_Fert: Shareweights of global fertilizer sector technologies
+    # Notes of the workflow: The pipeline below functions as the interpolate_and_melt
+    #                        function in the old data system. The pipeline first extracts the years
+    #                        needed in interpolation then constructs the layout with proper id tags (columns)
+    #                        for interpolation years, and finally performs interpolation using 'rule=1' as in
+    #                        old data system. For more information about 'rule' please see ?approx_fun.
+    #                        Same workflow applies at several places in this chunk.
+    A322.globaltech_shrwt %>%
+      gather(year, share.weight, matches(YEAR_PATTERN)) %>%
+      mutate(year = as.numeric(year)) ->
+      A322.globaltech_shrwt_long
+
+    df_years <- unique(A322.globaltech_shrwt_long$year)
+    int_years <- setdiff(c(BASE_YEARS, FUTURE_YEARS), df_years)
+
+    A322.globaltech_shrwt_long %>%
+      select(-share.weight, -year) %>%
+      unique() %>%
+      repeat_add_columns(tibble(year = int_years)) %>%
+      bind_rows(A322.globaltech_shrwt_long) %>%
+      group_by(supplysector, subsector, technology) %>%
+      mutate(share.weight = approx_fun(year, share.weight, rule = 1)) %>%
+      ungroup %>%
+      filter(year %in% c(BASE_YEARS, FUTURE_YEARS)) %>%
+      rename(sector.name = supplysector, subsector.name = subsector) ->
+      L2322.GlobalTechShrwt_Fert
+
+    # L2322.GlobalTechCoef_Fert: Energy inputs and coefficients of global fertilizer energy use and feedstocks technologies
+    A322.globaltech_coef %>%
+      gather(year, coefficient, matches(YEAR_PATTERN)) %>%
+      mutate(year = as.numeric(year)) ->
+      A322.globaltech_coef_long
+
+    df_years <- unique(A322.globaltech_coef_long$year)
+    int_years <- setdiff(c(BASE_YEARS, FUTURE_YEARS), df_years)
+
+    A322.globaltech_coef_long %>%
+      select(-coefficient, -year) %>%
+      unique() %>%
+      repeat_add_columns(tibble(year = int_years)) %>%
+      bind_rows(A322.globaltech_coef_long) %>%
+      group_by(supplysector, subsector, technology, minicam.energy.input) %>%
+      mutate(coefficient = approx_fun(year, coefficient, rule = 1)) %>%
+      mutate(coefficient = round(coefficient, energy.DIGITS_COEFFICIENT)) %>%
+      ungroup %>%
+      filter(year %in% c(BASE_YEARS, FUTURE_YEARS)) %>%
+      rename(sector.name = supplysector, subsector.name = subsector) ->
+      L2322.GlobalTechCoef_Fert
+
+    # Costs of global technologies
+    # L2322.GlobalTechCost_Fert: Non-energy costs of global fertilizer manufacturing technologies
+    L2322.GlobalTechCoef_Fert %>%
+      select(LEVEL2_DATA_NAMES[["GlobalTechYr"]]) %>%
+      mutate(minicam.non.energy.input = "non-energy") %>%
+      left_join(L1322.Fert_NEcost_75USDkgN_F, by = c('technology' = 'fuel') ) %>%
```

Comment:
Why no `left_join_error_no_match`? Are we expecting multiple matches or NAs?

Reply:
Yes. NAs are expected. Comments added.